airflow/providers/apache/hive/provider.yaml

```diff
@@ -56,6 +56,7 @@ dependencies:
   # the sasl library anyway (and there sasl library version is not relevant)
   - sasl>=0.3.1; python_version>="3.9"
   - thrift>=0.9.2
+  - impyla
```

Comment:
[impyla](https://github.com/cloudera/impyla) doesn't make any asyncio calls in execute_async.

Reply:
We have written our own asyncio method as shown below, because  impyla give us a handle immediately after submitting the query
 
```
async def partition_exists(self, table: str, schema: str, partition: str, polling_interval: float) -> str:
        """
        Checks for the existence of a partition in the given hive table.

        :param table: table in hive where the partition exists.
        :param schema: database where the hive table exists
        :param partition: partition to check for in given hive database and hive table.
        :param polling_interval: polling interval in seconds to sleep between checks
        """
        client = self.get_hive_client()
        cursor = client.cursor()
        query = f"show partitions {schema}.{table} partition({partition})"
        cursor.execute_async(query)
        while cursor.is_executing():
            await asyncio.sleep(polling_interval)
        results = cursor.fetchall()
        if len(results) == 0:
            return "failure"
        return "success"
```

Reply:
This unfortunetly not transform sync code into asyncio. If this kind transformation would be so easy than we had to transform any sync method to asyncio implementation.

So offhand most of `impyla` methods makes blocking io requests:
when you call `cursor.execute_async` as well as `cursor.is_executing()` and also `cursor.fetchall()`

Reply:
most sync method wait for the operation to complete, which is not what impyla does here. 

@kaxil any thoughts ?

Reply:
Not every async implementation is asyncio-compatible.

`asyncio` stand for Asynchronous I/O however `impyla` provides asynchronous execution with block I/O.

Implementation of `impyla` would be perfect for regular Sensor rather than Trigger and defer Operators